### PR TITLE
Pin rust version to 1.65 instead of pinning time-core

### PR DESCRIPTION
**Pull request by timsueberkrueb**
# Summary

See https://github.com/arlyon/async-stripe/pull/421#issuecomment-1683880259

### Checklist

- [x] ran `cargo make fmt`
- [x] using [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) to hightlight user-facing fixes and features

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -23,6 +23,7 @@ documentation = "https://docs.rs/async-stripe"
 keywords = ["stripe", "v1", "api", "async"]
 categories = ["api-bindings"]
 edition = "2021"
+rust-version = "1.65.0"
```
Comment (arlyon) [inline Cargo.toml:26]: The version is actually set above so you can remove this line
```suggestion

```

**Review by arlyon — CHANGES_REQUESTED** [review]

**Comment by arlyon** [comment]
looks like `time` now needs 1.67.0 so lets just bump our msrv to that while we are at it. change the version in cargo.toml and in the .github worflows (for CI)

**Comment by timsueberkrueb** [comment]
@arlyon done, thank you!

**Comment by arlyon** [comment]
Running CI again, TY

**Comment by timsueberkrueb** [comment]
@arlyon the CI failing is an unrelated issue I assume?

**Comment by timsueberkrueb** [comment]
Hey @arlyon, any update on this one? Thanks :)

**Comment by timsueberkrueb** [comment]
Hey @arlyon, it would be really cool if we could land this and have a new crates.io release containing the fix. Thanks!

**Comment by arlyon** [comment]
Fixed in this PR https://github.com/arlyon/async-stripe/pull/433

**Pull request closed without merging**